scripts/manageLAND.js

```diff
@@ -0,0 +1,206 @@
+            retryFailedTxs: options.retryFailedTxs,
+            from,
+            to,
+            landBatches
+          })
+        })
+      )
+  }
+}
+
+async function transferLANDs(args) {
+  const {
+    from,
+    to,
+    landBatches,
+    batchSize,
+    gasPrice,
+    shouldRetry,
+    txDelay
+  } = args
+
+  const account = eth.getAccount()
+  const LANDRegistryContract = eth.getContract('LANDRegistry')
+
+  const isApprovedForAll = await LANDRegistryContract.isApprovedForAll(
+    from,
+    account
+  )
+
+  log.info(`Transfering LANDs to ${to}`)
+
+  const txsToRetry = await executeTransactions(
+    landBatches,
+    recipientsBatch =>
+      recipientsBatch.map(async landBatch => {
+        const xs = []
+        const ys = []
+        let message = ''
+        const landBatchPromises = landBatch.map(async land => {
+          if (
+            !isApprovedForAll &&
+            from.toLowerCase() !== account.toLowerCase()
+          ) {
+            if (
+              (await LANDRegistryContract.getApproved(
+                land.id
+              )).toLowerCase() !== account.toLowerCase()
+            ) {
+              throw new Error(
+                `Error: ${account} does not have permission to move (${
+                  land.x
+                }, ${land.y})`
+              )
+            }
+          }
+          xs.push(land.x)
+          ys.push(land.y)
+          message += `Sending (${land.x}, ${land.y}) to ${to}\n`
+        })
+
+        await Promise.all(landBatchPromises)
+
+        log.info(message)
+        const hash = await LANDRegistryContract.transferManyLand(xs, ys, to, {
+          gasPrice: gasPrice,
+          from: account
+        })
+
+        return { hash, data: landBatch }
+      }),
+    { batchSize, txDelay }
+  )
+
+  log.info(`Sent ${landBatches.length - txsToRetry.length} transactions`)
+
+  if (txsToRetry.length > 0 && shouldRetry) {
+    const landBatchessToRetry = txsToRetry.map(tx => tx.data)
+    log.info(`Retrying on ${landBatchessToRetry.length} recipients`)
+    return transferLANDs(landBatchessToRetry, ...args.slice(1))
+  }
+}
+
+async function getBalance(from) {
+  return eth.getContract('LANDRegistry').balanceOf(from)
+}
+
+async function getLANDBatches(from, balance) {
+  log.info(`Getting ${balance} LAND ids to transfer...`)
+
+  const LANDRegistryContract = eth.getContract('LANDRegistry')
+  let index = 0
+  const landBatches = []
+
+  await asyncBatch({
+    elements: new Array(balance).fill(0),
+    callback: async fakeIndexes => {
+      const lands = []
+      const promises = fakeIndexes.map(async () => {
+        const id = await LANDRegistryContract.tokenOfOwnerByIndex(from, index++)
+        const [x, y] = await LANDRegistryContract.decodeTokenId(id)
+        lands.push({ x: x.toNumber(), y: y.toNumber(), id })
+      })
+
+      await Promise.all(promises)
+      landBatches.push(lands)
+    },
+    batchSize: 70, //Maximum LAND to be transferred in bulk
```

Comment:
This is not using the batchSize defined in the options, is it right?

Reply:
yes, I will add a new option for this anyways! good catch